static/gsApp/components/quotaExceededNavItem.tsx

```diff
@@ -0,0 +1,260 @@
+import type {Subscription} from 'getsentry/types';
+import {getCategoryInfoFromPlural} from 'getsentry/utils/billing';
+import {listDisplayNames, sortCategoriesWithKeys} from 'getsentry/utils/dataCategory';
+
+const ANIMATE_PROPS: MotionProps = {
+  animate: {
+    rotate: [0, -15, 15, -15, 15, -15, 0],
+    scale: [1, 1.25, 1.25, 1.25, 1.25, 1.25, 1],
+  },
+  transition: {
+    duration: 0.7,
+    repeat: Infinity,
+    repeatType: 'loop',
+    type: 'easeOut',
+    delay: 2,
+    repeatDelay: 1,
+  },
+};
+
+function QuotaExceededContent({
+  exceededCategories,
+  subscription,
+  organization,
+  onDismiss,
+  isDismissed,
+}: {
+  exceededCategories: string[];
+  isDismissed: boolean;
+  onDismiss: () => void;
+  organization: Organization;
+  subscription: Subscription;
+}) {
+  const eventTypes: EventType[] = exceededCategories.map(category => {
+    const categoryInfo = getCategoryInfoFromPlural(category as DataCategory);
+    return (categoryInfo?.plural ?? category) as EventType;
+  });
+  return (
+    <Container>
+      <Header>
+        <HeaderTitle>{t('Status')}</HeaderTitle>
+      </Header>
+      <Body>
+        <Title>{t('Quota Exceeded')}</Title>
+        <Description>
+          {tct(
+            'You’ve run out of [exceededCategories] for this billing cycle. This means we are no longer monitoring or ingesting events and showing them in Sentry.',
+            {
+              exceededCategories: listDisplayNames({
+                plan: subscription.planDetails,
+                categories: exceededCategories,
+                hadCustomDynamicSampling: subscription.hadCustomDynamicSampling,
+              }),
+            }
+          )}
+        </Description>
+        <ActionContainer>
+          <AddEventsCTA
+            organization={organization}
+            subscription={subscription}
+            buttonProps={{
+              size: 'xs',
+            }}
+            eventTypes={eventTypes}
+            notificationType="overage_critical"
+            referrer={`overage-alert-sidebar-${eventTypes.join('-')}`}
+            source="quota-overage"
+            handleRequestSent={() => {}}
+          />
+          <DismissContainer>
+            <Checkbox
+              name="dismiss"
+              checked={isDismissed}
+              disabled={isDismissed}
+              onChange={() => {
+                onDismiss();
+              }}
+            />
+            <CheckboxLabel>{t("Don't annoy me again")}</CheckboxLabel>
+          </DismissContainer>
+        </ActionContainer>
+      </Body>
+    </Container>
+  );
+}
+
+function PrimaryNavigationQuotaExceeded({
+  subscription,
+  organization,
+}: {
+  organization: Organization;
+  subscription: Subscription;
+}) {
+  const [exceededCategories, setExceededCategories] = useState<string[]>([]);
+  const [promptsToCheck, setPromptsToCheck] = useState<string[]>([]);
+  const [isDismissed, setIsDismissed] = useState(false);
```

Comment:
Sorry, found something else that looks a little off:

Are all these `useState`s necessary? It seems like you can calculate all these directly from the subscription and the return value of `usePrompts()`. It simplifies code a bunch if you can avoid storing duplicate data in state.

Reply:
ugh yeah you're right, i don't actually need these 🙏🏽